src/Microsoft.VisualStudio.Editors/PropPages/PackagePropPage.vb

```diff
@@ -230,7 +285,117 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
+                'I don't believe that undo will work with the license file text box because it is not user populated
+                SetLicenseRadioButtons(True)
+            End If
+        End Sub
+
+        'These GotFocus methods are for when the property page is first entered and neither have a value
+        'It would make sense to selected the corresponding radio button when the text box recieves focus
+        Private Sub PackageLicenseExpression_GotFocus(sender As Object, e As EventArgs) Handles PackageLicenseExpression.GotFocus
+            SetLicenseRadioButtons(True)
+        End Sub
+
+        Private Sub LicenseFileNameTextBox_GotFocus(sender As Object, e As EventArgs) Handles LicenseFileNameTextBox.GotFocus
+            SetLicenseRadioButtons(False)
+        End Sub
+
+        Private Sub LicenseBrowseButton_GotFocus(sender As Object, e As EventArgs) Handles LicenseBrowseButton.GotFocus
+            SetLicenseRadioButtons(False)
+        End Sub
+
+        Private Shared Function GetRelativePath(fullPath As String, basePath As String) As String
+            Dim baseUri = New Uri(basePath)
+            Dim fullUri = New Uri(fullPath)
+            Dim relativeUri = baseUri.MakeRelativeUri(fullUri)
+            Return relativeUri.ToString().Replace("/", "\")
+        End Function
+
+        Private Sub AddLicenseItemToProject(fileName As String)
+            Dim unconfiguredProject = GetUnconfiguredProject(ProjectHierarchy)
+            Dim configuredProject As ProjectSystem.ConfiguredProject = ThreadHelper.JoinableTaskFactory.Run(Function()
+                                                                                                                Return unconfiguredProject.GetSuggestedConfiguredProjectAsync
+                                                                                                            End Function)
+            Dim projectSourceItemProvider = configuredProject.Services.ExportProvider.GetExportedValue(Of ProjectSystem.IProjectSourceItemProvider)()
+            Dim correctDirectory = Directory.GetParent(unconfiguredProject.FullPath).ToString
+            Dim relativePath As String = GetRelativePath(Path.GetFullPath(fileName), correctDirectory + "\")
+            LicenseFileNameTextBox.Text = relativePath
+            'The TextBox needs to have the relative path, so the property isn't linked to the TextBox. It must be set manually.
+            SetCommonPropertyValue(GetPropertyDescriptor("PackageLicenseFile"), Path.GetFileName(fileName))
+            ThreadHelper.JoinableTaskFactory.Run(Function()
+                                                     Return projectSourceItemProvider.AddAsync("None", relativePath, {(New KeyValuePair(Of String, String)("Pack", "True")), New KeyValuePair(Of String, String)("PackagePath", "")})
+                                                 End Function)
+        End Sub
```

Comment:
This sub has two calls to `JTF.Run`. Making a single call (and doing the intermediate work within the lamda) would reduce the overhead.

Alternatively make `AddLicenseItemToProject` async and move the JTF call to its call in the click handler below.

Reply:
The dependency on CPS here is not needed, and will actually cause troubles - there is no guarantee that the tree has updated with the new item by the time this call returns. Please just use Project.ProjectItem which makes it project system agnostic and guarantees that the tree has been updated.

Reply:
I am not sure that this will work - Nuget Pack needs an item in the project file. It doesn't seem there is a way for Project.ProjectItem.AddItem does not generate an item in the project file. If it can generate the item, is there a way to change the metadata on the item? Pack also needs specific metadata. From a cursory glace, it seems like a ProjectItem's ProjectPropertiesContext does not have metadata. 

Also, the desired behavior is that the item is not copied to project.